tasking/pulp/tasking/base.py

```diff
@@ -141,16 +142,19 @@ def delete_worker(name, normal_shutdown=False):
         msg = msg % {'name': name}
         _logger.error(msg)
 
-    worker = Worker.objects.get(name=name)
-
-    # Cancel all of the tasks that were assigned to this worker's queue
-    for task_status in worker.tasks.filter(state__in=TASK_INCOMPLETE_STATES):
-        cancel(task_status.pk)
+    try:
+        worker = Worker.objects.get(name=name)
```

Comment:
What situation occurs where `delete_worker` is called yet the worker does not exist?